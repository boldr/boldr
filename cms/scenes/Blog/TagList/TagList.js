import React, { Component, PropTypes } from 'react';
import { provideHooks } from 'redial';
import { connect } from 'react-redux';
import { Loader, Grid, Col, Row } from '../../../components';
import TagListCard from './TagListCard';
import { requestPostTags } from './actions';

@provideHooks({
  fetch: ({ dispatch, params: { id } }) => dispatch(requestPostTags(id))
})
class TagList extends Component {
  static propTypes = {
    tags: PropTypes.object.isRequired
  }
  render() {
    return (
      <div>
        <Grid fluid>
        <Row>
        {
          this.props.tags.isLoading ?
          <Loader /> :
          this.props.tags.posts.map(post =>
            <Col key={ post.id } xs={ 12 } md={ 4 }>
              <TagListCard { ...post } />
            </Col>)
        }
        </Row>
        </Grid>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    tags: state.tags,
    isLoading: state.tags.isLoading,
    posts: state.tags.posts
  };
};

export default connect(mapStateToProps)(TagList);
