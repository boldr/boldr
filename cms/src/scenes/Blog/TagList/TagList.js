/* @flow */

import React, { Component, PropTypes } from 'react';
import { provideHooks } from 'redial';
import { connect } from 'react-redux';
import { Loader, Grid, Col, Row } from 'components';
import { requestPostTags } from 'state/dux/tag';
import TagListCard from './TagListCard';

export type Props = {tags?: Object};

@provideHooks({
  fetch: ({ dispatch, params: { name } }) => dispatch(requestPostTags(name))
})
class TagList extends Component {
  props: Props;
  render() {
    return (
      <div>
        <Grid fluid>
        <Row>
        {
          this.props.tags.isLoading ?
          <Loader /> :
          this.props.tags.posts.map(post =>
            <Col key={ post.id } xs={ 12 } md={ 4 }>
              <TagListCard { ...post } />
            </Col>)
        }
        </Row>
        </Grid>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    tags: state.tags,
    isLoading: state.tags.isLoading,
    posts: state.tags.posts
  };
};

export default connect(mapStateToProps)(TagList);
