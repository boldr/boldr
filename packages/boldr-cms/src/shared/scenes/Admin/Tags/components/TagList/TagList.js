/* @flow */

import React, { PropTypes } from 'react';
import ListItem from 'react-md/lib/Lists/ListItem';
import Paper from 'react-md/lib/Papers';
import Avatar from 'react-md/lib/Avatars';
import FontIcon from 'react-md/lib/FontIcons';

type Props = {
  tags: Array<Tag>,
  handleTagClick: Function,
  handleDeleteTagClick: Function,
};

const TagList = (props: Props) => {
  function handleClick(tag: Object) {
    props.handleTagClick(tag);
  }
  function handleClickDelete(tag) {
    const tagId = tag.id;

    props.handleDeleteTagClick(tagId);
  }
  return (
    <div>
      {props.tags.map(tag => (
        <ListItem
          key={ tag.id }
          primaryText={ tag.name }
          rightIcon={ <FontIcon onClick={ () => handleClickDelete(tag) }>delete_forever</FontIcon> }
          secondaryText={ tag.description }
          onClick={ () => handleClick(tag) }
        />
      ))}
    </div>
  );
};

export default TagList;
