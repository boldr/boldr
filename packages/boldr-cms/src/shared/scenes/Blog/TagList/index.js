import TagListContainer from './TagListContainer';

export default TagListContainer;
