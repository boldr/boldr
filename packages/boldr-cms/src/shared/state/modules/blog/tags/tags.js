import { combineReducers } from 'redux';
import { FETCH_POSTS_SUCCESS } from '../posts/constants';
import * as t from './constants';

export const STATE_KEY = 'tags';

const all = (state = {}, action) => {
  switch (action.type) {
    case t.FETCH_TAGS_SUCCESS:
    case FETCH_POSTS_SUCCESS:
      return {
        ...state,
        ...action.payload.entities.tags,
      };
    default:
      return state;
  }
};

const ids = (state = [], action) => {
  switch (action.type) {
    case t.FETCH_TAGS_SUCCESS:
      return action.payload.result;
    default:
      return state;
  }
};


const isFetching = (state = false, action) => {
  switch (action.type) {
    case t.FETCH_TAGS_REQUEST:
    case t.FETCH_TAGGED_POST_REQUEST:
      return true;
    case t.FETCH_TAGS_SUCCESS:
    case t.FETCH_TAGS_FAILURE:
    case t.FETCH_TAGGED_POST_SUCCESS:
    case t.FETCH_TAGGED_POST_FAILURE:
      return false;
    default:
      return state;
  }
};

const currentTag = (state = {}, action) => {
  switch (action.type) {
    case t.FETCH_TAGGED_POST_SUCCESS:
      return {
        ...state,
        ...action.payload,
      };
    case t.SELECT_TAG:
      return {
        ...state,
        ...action.tag,
      };

    default:
      return state;
  }
};
export default combineReducers({
  all,
  ids,
  isFetching,
  currentTag,
});
