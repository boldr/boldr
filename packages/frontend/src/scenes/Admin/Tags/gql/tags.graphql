query getTags($offset: Int!, $limit: Int!) {
  getTags(offset: $offset, limit: $limit) {
    id
    name
    articles {
      title
      slug
      id
    }
  }
}
