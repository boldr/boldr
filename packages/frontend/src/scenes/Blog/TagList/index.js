// @flow
import React from 'react';
import { graphql } from 'react-apollo';
import universal from 'react-universal-component';
import ARTICLES_FOR_TAG from '../gql/articlesForTag.graphql';
import type { ArticlesType, MatchParams } from '../../../types/boldr';

const UniversalTagList = universal(import('./TagList'));

type Props = {
  loading: boolean,
  error?: Object,
  getArticlesForTag: ArticlesType,
  match: MatchParams,
};

const TagList = ({ loading, error, match, getArticlesForTag }: Props) =>
  <UniversalTagList isLoading={loading} error={error} match={match} articles={getArticlesForTag} />;

// $FlowIssue
export default graphql(ARTICLES_FOR_TAG, {
  options: props => ({
    variables: {
      name: props.match.params.name,
    },
  }),
  props: ({ ownProps: { match }, data: { loading, error, getArticlesForTag } }) => ({
    loading,
    error,
    match,
    getArticlesForTag,
  }),
})(TagList);
