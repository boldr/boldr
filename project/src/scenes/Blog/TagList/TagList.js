/* @flow */

import React from 'react';
import Helmet from 'react-helmet';
import styled from 'styled-components';
import { Section } from '@boldr/ui/Layout';
import Heading from '@boldr/ui/Heading';
import Loader from '@boldr/ui/Loader';
import type { ArticlesType, MatchParams } from '../../../types/boldr';
import ArticleCard from '../components/ArticleCard';

type Props = {
  isLoading: boolean,
  articles: ArticlesType,
  match: MatchParams,
};

const Container = styled.div`
  flex: 1;
  max-width: 1034px;
  width: 100%;
  margin: 30px auto 100px;
  display: flex;
  flex-wrap: wrap;
`;
const TagList = (props: Props) => {
  const { isLoading, articles, match: { params } } = props;
  if (isLoading) {
    return <Loader />;
  }
  if (!articles) {
    return <h1>No matching posts</h1>;
  }
  return (
    <div>
      <Helmet title={`Posts tagged ${params.name}`} />
      <Section>
        <Heading type="h2" text={params.name} />
      </Section>
      <Container>
        {articles ? (
          articles.map(article => <ArticleCard key={article.id} featured={false} {...article} />)
        ) : (
          <Loader />
        )}
      </Container>
    </div>
  );
};

export default TagList;
