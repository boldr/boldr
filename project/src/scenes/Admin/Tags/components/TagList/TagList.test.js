import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import tags from '../../__fixtures__/tags.fixtures';
import TagList from './TagList';

describe('<TagList /> component', () => {
  it('<TagList />, renders the tag list with props', () => {
    const wrapper = shallow(<TagList tags={tags} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
