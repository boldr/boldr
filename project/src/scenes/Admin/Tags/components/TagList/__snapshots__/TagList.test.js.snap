// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagList /> component <TagList />, renders the tag list with props 1`] = `
<TagList__List>
  <Link
    key="1"
    replace={false}
    to="/admin/tags/javascript"
  >
    <TagList__ListItem>
      <TagList__InnerList>
        <TagList__InnerListItem>
          <TagList__TagName>
            javascript
          </TagList__TagName>
        </TagList__InnerListItem>
        <TagList__InnerListItem>
          <TagList__TagDesc>
            0
          </TagList__TagDesc>
        </TagList__InnerListItem>
        <TagList__InnerListItem>
          <Trash
            fill="#222"
            name="Trash"
            onClick={[Function]}
            size={24}
          />
        </TagList__InnerListItem>
      </TagList__InnerList>
    </TagList__ListItem>
  </Link>
  <Link
    key="2"
    replace={false}
    to="/admin/tags/apple"
  >
    <TagList__ListItem>
      <TagList__InnerList>
        <TagList__InnerListItem>
          <TagList__TagName>
            apple
          </TagList__TagName>
        </TagList__InnerListItem>
        <TagList__InnerListItem>
          <TagList__TagDesc>
            0
          </TagList__TagDesc>
        </TagList__InnerListItem>
        <TagList__InnerListItem>
          <Trash
            fill="#222"
            name="Trash"
            onClick={[Function]}
            size={24}
          />
        </TagList__InnerListItem>
      </TagList__InnerList>
    </TagList__ListItem>
  </Link>
  <Link
    key="3"
    replace={false}
    to="/admin/tags/Skiing"
  >
    <TagList__ListItem>
      <TagList__InnerList>
        <TagList__InnerListItem>
          <TagList__TagName>
            Skiing
          </TagList__TagName>
        </TagList__InnerListItem>
        <TagList__InnerListItem>
          <TagList__TagDesc>
            0
          </TagList__TagDesc>
        </TagList__InnerListItem>
        <TagList__InnerListItem>
          <Trash
            fill="#222"
            name="Trash"
            onClick={[Function]}
            size={24}
          />
        </TagList__InnerListItem>
      </TagList__InnerList>
    </TagList__ListItem>
  </Link>
</TagList__List>
`;
